rl4co/envs/routing/mtvrp/env.py

```diff
@@ -349,9 +349,14 @@ def check_solution_validity(td: TensorDict, actions: torch.Tensor):
             curr_time = torch.max(
                 curr_time + dist, gather_by_index(td["time_windows"], next_node)[..., 0]
             )
+
+            new_shape = curr_time.size()
+            skip_open_end = td["open_route"].view(*new_shape) & (next_node == 0).view(*new_shape)
```

Comment:
Makes sense, good catch.
Anyways, I recommend setting `check_solution` to `False` when training; otherwise, the solution will be checked at each step and it can be a bit slow. I will add a warning

Reply:
Actually, I don't think this is necessary. Since `skip_open_end` will only be `true` if `next_node == 0`, and since the depot has the highest time window end, `curr_time <= gather_by_index(td["time_windows"], next_node)[..., 1])` should always be `True` *except* when `curr_time` is very close to the max time already and then the duration in that last node is long enough to go over the time limit - is that something we want to allow?

Reply:
I agree with ngastzepeda as the last node in the route should also satisfy the time window constraints (allow it back it depot even when it is OVRP). However, I find some outliners when training the MTVRP (i.e., the time window of the last node of OVRP route may exceed the max time window). I do not yet know the exact reason, instance generation, or masking procedure.

Reply:
I will have a check.